From 6765c9d66f4ae89bfd8a2b17683123a58103c8a3 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Yilli=C3=A9?= <yilliee@protonmail.com>
Date: Sun, 3 Oct 2021 18:39:40 +0500
Subject: [PATCH 1/2] Add an option to specify a custom x-axis value for status bar
 items
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

 - The var TW_CUSTOM_CLOCK_POS, TW_CUSTOM_CPU_TEMP_POS and
   TW_CUSTOM_BATTERY_POS can be used to specify a custom position
   for the clock, cpu temperature and battery percentage on the x-axis.
 - One of the use-cases of this var is when the clock overlaps with the
   camera cutout. In that case, instead of specifiying an offset for
   the whole gui on y-axis, the statsu bar items can be offset along the
   x-axis only.

Signed-off-by: Yillié <yilliee@protonmail.com>
---
 Android.mk                      | 8 +++++++-
 gui/theme/landscape_hdpi/ui.xml | 9 ++++++---
 gui/theme/landscape_mdpi/ui.xml | 9 ++++++---
 gui/theme/portrait_hdpi/ui.xml  | 9 ++++++---
 gui/theme/portrait_mdpi/ui.xml  | 9 ++++++---
 gui/theme/watch_mdpi/ui.xml     | 9 ++++++---
 6 files changed, 37 insertions(+), 16 deletions(-)

diff --git a/Android.mk b/Android.mk
index f27ac4b8..9727c5f0 100755
--- a/Android.mk
+++ b/Android.mk
@@ -570,7 +570,13 @@ TW_THEME_VERSION := $(shell grep TW_THEME_VERSION bootable/recovery/variables.h
 
 LOCAL_POST_INSTALL_CMD += \
     sed -i "s/{themeversion}/$(TW_THEME_VERSION)/" $(TARGET_RECOVERY_ROOT_OUT)/twres/splash.xml; \
-    sed -i "s/{themeversion}/$(TW_THEME_VERSION)/" $(TARGET_RECOVERY_ROOT_OUT)/twres/ui.xml;
+    sed -i "s/{themeversion}/$(TW_THEME_VERSION)/" $(TARGET_RECOVERY_ROOT_OUT)/twres/ui.xml; \
+    if [ -n "$(TW_CUSTOM_BATTERY_POS)" ]; then sed -i "s/{battery_pos}/$(TW_CUSTOM_BATTERY_POS)/"  $(TARGET_RECOVERY_ROOT_OUT)/twres/ui.xml; \
+    else sed -i "s/{battery_pos}/%indent_right%/"  $(TARGET_RECOVERY_ROOT_OUT)/twres/ui.xml; fi; \
+    if [ -n "$(TW_CUSTOM_CLOCK_POS)" ]; then sed -i "s/{clock_pos}/$(TW_CUSTOM_CLOCK_POS)/"  $(TARGET_RECOVERY_ROOT_OUT)/twres/ui.xml; \
+    else sed -i "s/{clock_pos}/%center_x%/"  $(TARGET_RECOVERY_ROOT_OUT)/twres/ui.xml; fi; \
+    if [ -n "$(TW_CUSTOM_CPU_POS)" ]; then sed -i "s/{cpu_pos}/$(TW_CUSTOM_CPU_POS)/"  $(TARGET_RECOVERY_ROOT_OUT)/twres/ui.xml; \
+    else sed -i "s/{cpu_pos}/%indent%/"  $(TARGET_RECOVERY_ROOT_OUT)/twres/ui.xml; fi;
 
 include $(BUILD_EXECUTABLE)
 
diff --git a/gui/theme/landscape_hdpi/ui.xml b/gui/theme/landscape_hdpi/ui.xml
index bd055bd5..2f210f3e 100755
--- a/gui/theme/landscape_hdpi/ui.xml
+++ b/gui/theme/landscape_hdpi/ui.xml
@@ -226,6 +226,9 @@
 		<variable name="pattern_dot_dia" value="32"/>
 		<variable name="pattern_line_w" value="12"/>
 		<variable name="pattern_size" value="648"/>
+		<variable name="tw_clock_pos_x" value="{clock_pos}"/>
+		<variable name="tw_cpu_pos_x" value="{cpu_pos}"/>
+		<variable name="tw_battery_pos_x" value="{battery_pos}"/>
 	</variables>
 
 	<mousecursor>
@@ -275,13 +278,13 @@
 			<text color="%text_color%">
 				<condition var1="tw_no_cpu_temp" var2="0"/>
 				<font resource="font_m"/>
-				<placement x="%indent%" y="%row1_header_y%"/>
+				<placement x="%tw_cpu_pos_x%" y="%row1_header_y%"/>
 				<text>{@cpu_temp=CPU: %tw_cpu_temp% °C}</text>
 			</text>
 
 			<text color="%text_color%">
 				<font resource="font_m"/>
-				<placement x="%center_x%" y="%row1_header_y%" placement="5"/>
+				<placement x="%tw_clock_pos_x%" y="%row1_header_y%" placement="5"/>
 				<text>%tw_time%</text>
 			</text>
 
@@ -292,7 +295,7 @@
 					<condition var1="tw_battery" op="&lt;" var2="101"/>
 				</conditions>
 				<font resource="font_m"/>
-				<placement x="%indent_right%" y="%row1_header_y%" placement="1"/>
+				<placement x="%tw_battery_pos_x%" y="%row1_header_y%" placement="1"/>
 				<text>{@battery_pct=Battery: %tw_battery%}</text>
 			</text>
 
diff --git a/gui/theme/landscape_mdpi/ui.xml b/gui/theme/landscape_mdpi/ui.xml
index 9ab8863d..b453c7a6 100755
--- a/gui/theme/landscape_mdpi/ui.xml
+++ b/gui/theme/landscape_mdpi/ui.xml
@@ -226,6 +226,9 @@
 		<variable name="pattern_dot_dia" value="12"/>
 		<variable name="pattern_line_w" value="4"/>
 		<variable name="pattern_size" value="252"/>
+		<variable name="tw_clock_pos_x" value="{clock_pos}"/>
+                <variable name="tw_cpu_pos_x" value="{cpu_pos}"/>
+                <variable name="tw_battery_pos_x" value="{battery_pos}"/>
 	</variables>
 
 	<mousecursor>
@@ -275,13 +278,13 @@
 			<text color="%text_color%">
 				<condition var1="tw_no_cpu_temp" var2="0"/>
 				<font resource="font_m"/>
-				<placement x="%indent%" y="%row1_header_y%"/>
+				<placement x="%tw_cpu_pos_x%" y="%row1_header_y%"/>
 				<text>{@cpu_temp=CPU: %tw_cpu_temp% °C}</text>
 			</text>
 
 			<text color="%text_color%">
 				<font resource="font_m"/>
-				<placement x="%center_x%" y="%row1_header_y%" placement="5"/>
+				<placement x="%tw_clock_pos_x%" y="%row1_header_y%" placement="5"/>
 				<text>%tw_time%</text>
 			</text>
 
@@ -292,7 +295,7 @@
 					<condition var1="tw_battery" op="&lt;" var2="101"/>
 				</conditions>
 				<font resource="font_m"/>
-				<placement x="%indent_right%" y="%row1_header_y%" placement="1"/>
+				<placement x="%tw_battery_pos_x%" y="%row1_header_y%" placement="1"/>
 				<text>{@battery_pct=Battery: %tw_battery%}</text>
 			</text>
 
diff --git a/gui/theme/portrait_hdpi/ui.xml b/gui/theme/portrait_hdpi/ui.xml
index f7d21da3..3d04b450 100755
--- a/gui/theme/portrait_hdpi/ui.xml
+++ b/gui/theme/portrait_hdpi/ui.xml
@@ -223,6 +223,9 @@
 		<variable name="pattern_dot_dia" value="32"/>
 		<variable name="pattern_line_w" value="12"/>
 		<variable name="pattern_size" value="828"/>
+		<variable name="tw_clock_pos_x" value="{clock_pos}"/>
+                <variable name="tw_cpu_pos_x" value="{cpu_pos}"/>
+                <variable name="tw_battery_pos_x" value="{battery_pos}"/>
 	</variables>
 
 	<mousecursor>
@@ -272,13 +275,13 @@
 			<text color="%text_color%">
 				<condition var1="tw_no_cpu_temp" var2="0"/>
 				<font resource="font_m"/>
-				<placement x="%indent%" y="%row1_header_y%"/>
+				<placement x="%tw_cpu_pos_x%" y="%row1_header_y%"/>
 				<text>{@cpu_temp=CPU: %tw_cpu_temp% °C}</text>
 			</text>
 
 			<text color="%text_color%">
 				<font resource="font_m"/>
-				<placement x="%center_x%" y="%row1_header_y%" placement="5"/>
+				<placement x="%tw_clock_pos_x%" y="%row1_header_y%" placement="5"/>
 				<text>%tw_time%</text>
 			</text>
 
@@ -289,7 +292,7 @@
 					<condition var1="tw_battery" op="&lt;" var2="101"/>
 				</conditions>
 				<font resource="font_m"/>
-				<placement x="%indent_right%" y="%row1_header_y%" placement="1"/>
+				<placement x="%tw_battery_pos_x%" y="%row1_header_y%" placement="1"/>
 				<text>{@battery_pct=Battery: %tw_battery%}</text>
 			</text>
 
diff --git a/gui/theme/portrait_mdpi/ui.xml b/gui/theme/portrait_mdpi/ui.xml
index 9189df0c..dad50cb7 100755
--- a/gui/theme/portrait_mdpi/ui.xml
+++ b/gui/theme/portrait_mdpi/ui.xml
@@ -223,6 +223,9 @@
 		<variable name="pattern_dot_dia" value="12"/>
 		<variable name="pattern_line_w" value="4"/>
 		<variable name="pattern_size" value="360"/>
+		<variable name="tw_clock_pos_x" value="{clock_pos}"/>
+                <variable name="tw_cpu_pos_x" value="{cpu_pos}"/>
+                <variable name="tw_battery_pos_x" value="{battery_pos}"/>
 	</variables>
 
 	<mousecursor>
@@ -272,13 +275,13 @@
 			<text color="%text_color%">
 				<condition var1="tw_no_cpu_temp" var2="0"/>
 				<font resource="font_m"/>
-				<placement x="%indent%" y="%row1_header_y%"/>
+				<placement x="%tw_cpu_pos_x%" y="%row1_header_y%"/>
 				<text>{@cpu_temp=CPU: %tw_cpu_temp% °C}</text>
 			</text>
 
 			<text color="%text_color%">
 				<font resource="font_m"/>
-				<placement x="%center_x%" y="%row1_header_y%" placement="5"/>
+				<placement x="%tw_clock_pos_x%" y="%row1_header_y%" placement="5"/>
 				<text>%tw_time%</text>
 			</text>
 
@@ -289,7 +292,7 @@
 					<condition var1="tw_battery" op="&lt;" var2="101"/>
 				</conditions>
 				<font resource="font_m"/>
-				<placement x="%indent_right%" y="%row1_header_y%" placement="1"/>
+				<placement x="%tw_battery_pos_x%" y="%row1_header_y%" placement="1"/>
 				<text>{@battery_pct=Battery: %tw_battery%}</text>
 			</text>
 
diff --git a/gui/theme/watch_mdpi/ui.xml b/gui/theme/watch_mdpi/ui.xml
index ed7ac825..76b40e75 100644
--- a/gui/theme/watch_mdpi/ui.xml
+++ b/gui/theme/watch_mdpi/ui.xml
@@ -182,6 +182,9 @@
 		<variable name="pattern_dot_dia" value="12"/>
 		<variable name="pattern_line_w" value="4"/>
 		<variable name="pattern_size" value="200"/>
+		<variable name="tw_clock_pos_x" value="{clock_pos}"/>
+                <variable name="tw_cpu_pos_x" value="{cpu_pos}"/>
+                <variable name="tw_battery_pos_x" value="{battery_pos}"/>
 	</variables>
 
 	<templates>
@@ -211,13 +214,13 @@
 			<text color="%text_color%">
 				<condition var1="tw_no_cpu_temp" var2="0"/>
 				<font resource="font_m"/>
-				<placement x="%indent%" y="%row1_header_y%"/>
+				<placement x="%tw_battery_pos_x%" y="%row1_header_y%"/>
 				<text>{@cpu_temp=CPU: %tw_cpu_temp% °C}</text>
 			</text>
 
 			<text color="%text_color%">
 				<font resource="font_m"/>
-				<placement x="%center_x%" y="%row1_header_y%" placement="5"/>
+				<placement x="%tw_clock_pos_x%" y="%row1_header_y%" placement="5"/>
 				<text>%tw_time%</text>
 			</text>
 
@@ -228,7 +231,7 @@
 					<condition var1="tw_battery" op="&lt;" var2="101"/>
 				</conditions>
 				<font resource="font_m"/>
-				<placement x="%indent_right%" y="%row1_header_y%" placement="1"/>
+				<placement x="%tw_battery_pos_x%" y="%row1_header_y%" placement="1"/>
 				<text>{@battery_pct=Battery: %tw_battery%}</text>
 			</text>
 
-- 
2.36.1

